### ISSUE TITLE
`limit_exceeded` is off by one in Redis backend

### ISSUE THREAD

**Issue by twolfson** [inveniosoftware/flask-ratelimiter#20]
While researching this module, I also found some non-security related issues. For example, `limit_exceeded` is reporting that we are over the limit by 1 request too early.

The simplest example is an endpoint that is rate limited to 1 request per 1 second.

```python
@ratelimit(limit=1, per=1)
def one_second():
    return 'hello'
```

We would expect the first request to go through and subsequent ones to fail. However, the first one fails because the `incr` has already run and we are using a `>=`. So it's `1 >= 1` which is true and as a result, we get a 429. To patch this, we can move to a `>`.

https://github.com/inveniosoftware/flask-ratelimiter/blob/v0.2.0/flask_ratelimiter/backends/simpleredis_backend.py#L57

Alternatively, it is quite wasteful to always run `incr`, even when we have exceeded the limit. I suggest moving to a `.get()` and runnning our `limit_exceeded` check before performing `incr`.

In either case, I am willing to open a PR with a fix. Thoughts/preferences?

**Comment by jirikuncar** [comment]
> Alternatively, it is quite wasteful to always run `incr`, even when we have exceeded the limit. I suggest moving to a `.get()` and runnning our `limit_exceeded` check before performing `incr`.

I like this one more. Can you also help us by adding correct test case? Thank you in advance.

**Comment by twolfson** [comment]
Sure thing. One PR coming up =)

**Comment by jirikuncar** [comment]
Closed via #21.
